private/buf/bufcheck/internal/bufcheckserver/internal/bufcheckserverhandle/lint.go

```diff
@@ -178,18 +186,1073 @@ func handleLintDirectorySamePackage(
+) error {
+	return handleLintPackageSameOptionValue(
+		responseWriter,
+		pkg,
+		pkgFiles,
+		func(file bufprotosource.File) string {
+			return strconv.FormatBool(file.JavaMultipleFiles())
+		},
+		bufprotosource.File.JavaMultipleFilesLocation,
+		"java_multiple_files",
+	)
+}
+
+func handleLintPackageSameJavaPackage(
+	responseWriter bufcheckserverutil.ResponseWriter,
+	_ bufcheckserverutil.Request,
+	pkg string,
+	pkgFiles []bufprotosource.File,
+) error {
+	return handleLintPackageSameOptionValue(
+		responseWriter,
+		pkg,
+		pkgFiles,
+		bufprotosource.File.JavaPackage,
+		bufprotosource.File.JavaPackageLocation,
+		"java_package",
+	)
+}
+
+func handleLintPackageSamePhpNamespace(
+	responseWriter bufcheckserverutil.ResponseWriter,
+	_ bufcheckserverutil.Request,
+	pkg string,
+	pkgFiles []bufprotosource.File,
+) error {
+	return handleLintPackageSameOptionValue(
+		responseWriter,
+		pkg,
+		pkgFiles,
+		bufprotosource.File.PhpNamespace,
+		bufprotosource.File.PhpNamespaceLocation,
+		"php_namespace",
+	)
+}
+
+func handleLintPackageSameRubyPackage(
+	responseWriter bufcheckserverutil.ResponseWriter,
+	_ bufcheckserverutil.Request,
+	pkg string,
+	pkgFiles []bufprotosource.File,
+) error {
+	return handleLintPackageSameOptionValue(
+		responseWriter,
+		pkg,
+		pkgFiles,
+		bufprotosource.File.RubyPackage,
+		bufprotosource.File.RubyPackageLocation,
+		"ruby_package",
+	)
+}
+
+func handleLintPackageSameSwiftPrefix(
+	responseWriter bufcheckserverutil.ResponseWriter,
+	_ bufcheckserverutil.Request,
+	pkg string,
+	pkgFiles []bufprotosource.File,
+) error {
+	return handleLintPackageSameOptionValue(
+		responseWriter,
+		pkg,
+		pkgFiles,
+		bufprotosource.File.SwiftPrefix,
+		bufprotosource.File.SwiftPrefixLocation,
+		"swift_prefix",
+	)
+}
+
+func handleLintPackageSameOptionValue(
+	responseWriter bufcheckserverutil.ResponseWriter,
+	pkg string,
+	pkgFiles []bufprotosource.File,
+	getFileOptionValue func(bufprotosource.File) string,
+	getFileOptionLocation func(bufprotosource.File) bufprotosource.Location,
+	name string,
+) error {
+	optionValueMap := make(map[string]struct{})
+	for _, file := range pkgFiles {
+		optionValueMap[getFileOptionValue(file)] = struct{}{}
+	}
+	if len(optionValueMap) > 1 {
+		_, noOptionValue := optionValueMap[""]
+		delete(optionValueMap, "")
+		optionValues := slicesext.MapKeysToSortedSlice(optionValueMap)
+		for _, file := range pkgFiles {
+			var message string
+			if noOptionValue {
+				message = fmt.Sprintf(
+					"Files in package %q have both values %q and no value for option %q and all values must be equal.",
+					pkg,
+					strings.Join(optionValues, ","),
+					name,
+				)
+			} else {
+				message = fmt.Sprintf(
+					"Files in package %q have multiple values %q for option %q and all values must be equal.",
+					pkg,
+					strings.Join(optionValues, ","),
+					name,
+				)
+			}
+			// TODO: investigate the case where noOptionValue is false but fileOptionLocation is nil.
```

Comment:
Yes this is strange. Perhaps the handlers should not assume these locations are not `nil`.